lib/reek/source/source_code.rb

```diff
@@ -37,9 +37,10 @@ def initialize(code, description, parser = Parser::Ruby22)
       # @return an instance of SourceCode
       def self.from(source)
         case source
-        when File   then new(source.read, source.path)
-        when IO     then new(source.readlines.join, 'STDIN')
-        when String then new(source, 'string')
+        when File     then new(source.read, source.path)
+        when IO       then new(source.readlines.join, 'STDIN')
+        when Pathname then new(source.read, source.to_s)
```

Comment:
I didn't really pay much attention to this PR, but perhaps I should have.
I'd really like to not add more options to `.from`. I find the fact that Pathname is also apparently something to read from rather confusing; It is sort of similar to File, but not really.